sentry/src/main/java/io/sentry/Baggage.java

```diff
@@ -18,79 +27,122 @@ public final class Baggage {
   static final @NotNull String CHARSET = "UTF-8";
   static final @NotNull Integer MAX_BAGGAGE_STRING_LENGTH = 8192;
   static final @NotNull Integer MAX_BAGGAGE_LIST_MEMBER_COUNT = 64;
+  static final @NotNull String SENTRY_BAGGAGE_PREFIX = "sentry-";
 
   final @NotNull Map<String, String> keyValues;
+  final @Nullable String thirdPartyHeader;
+  private boolean mutable;
   final @NotNull ILogger logger;
 
+  @NotNull
   public static Baggage fromHeader(
       final @Nullable List<String> headerValues, final @NotNull ILogger logger) {
-    final Map<String, String> keyValues = new HashMap<>();
+    return Baggage.fromHeader(headerValues, false, logger);
+  }
+
+  @NotNull
+  public static Baggage fromHeader(
+      final @Nullable List<String> headerValues,
+      final boolean includeThirdPartyValues,
```

Comment:
Why do we need the `includeThirdPartyValues`? we always need to merge and keep the 3rd party values.

Reply:
For incoming `baggage` headers (server side) we don't want to save the third party entries. With the `includeThirdPartyValues` flag we can achieve this. There was some recent discussion about this internally.

Reply:
Oh yeah indeed, that makes sense, we don't handle incoming requests on Mobile so I was unsure, thanks.